i2c/ti/ads11xx.go

```diff
@@ -0,0 +1,221 @@
+// Package ti contains drivers for IC's produces by Texas Instruments.
+package ti
+
+import (
+	"fmt"
+	"math"
+
+	"golang.org/x/exp/io/i2c"
+)
+
+type dataRate struct {
+	// sps is the data rate samples per second.
+	sps int
+
+	bitMask int
+
+	// size is the amount of bits that represent the output code
+	size uint
+}
+
+type ads11xx struct {
+	Conn *i2c.Device
+	Vref float64
+
+	dataRate dataRate
+	pga      int
+
+	// dataRates is a map that holds all valid values for data rate.
+	dataRates []dataRate
+}
+
+func newADS11xx(conn *i2c.Device, vref float64, dataRate, pga int, dataRates []dataRate) (ads11xx, error) {
+	a := ads11xx{
+		Conn:      conn,
+		Vref:      vref,
+		dataRates: dataRates,
+	}
+
+	if err := a.setDataRate(dataRate); err != nil {
+		return a, err
+	}
+
+	if err := a.SetPGA(pga); err != nil {
+		return a, err
+	}
+
+	return a, nil
+}
+
+// Voltage queries the channel of an ADC and returns its voltage.
+func (a ads11xx) Voltage(channel int) (float64, error) {
+	code, err := a.OutputCode(channel)
+	if err != nil {
+		return 0, err
+	}
+
+	max := math.Pow(2, float64(a.dataRate.size))
+	return ((a.Vref / max) * float64(code) / float64(a.pga)), nil
+}
+
+// OutputCode queries the channel and returns its digital output code. The
+// maximum code depends on the selected data rate. How higher the data rate,
+// how lower the amount of bits used.
+func (a ads11xx) OutputCode(channel int) (int, error) {
+	if channel != 1 {
+		return 0, fmt.Errorf("channel %d is invalid, ADC has only 1 channel", channel)
+	}
+
+	in := make([]byte, 2)
+	if err := a.Conn.Read(in); err != nil {
+		return 0, fmt.Errorf("failed to read output code: %v", err)
+	}
+
+	msb := in[0] & byte(math.Pow(2, float64(a.dataRate.size-8))-1)
+	v := (int(msb) << 8) + int(in[1])
+
+	return v, nil
+}
+
+// PGA reads the config register of the ADS1100 and returns the current PGA.
```

Comment:
I think `ADS1100` should be  `ADS11xx` because this method is still part of the ADC11xx.